cwt/cose_message.py

```diff
@@ -309,16 +309,19 @@ def detach_payload(self: Self) -> Tuple[COSEMessage, bytes]:
         Detach a payload from the COSE message
 
         Returns:
-            Tuple[COSEMessage, bytes]: A byte string of the encoded COSE or a
-                cbor2.CBORTag object, and a byte string of the detached payload.
+            Tuple[COSEMessage, bytes]: The COSE message (self),
+            and a byte string of the detached payload.
         Raises:
             ValueError: The payload does not exist.
         """
 
         if not isinstance(self._payload, bytes):
```

Comment:
For now, if you want to implement holding the detached payload in a COSEMessage structure:

```suggestion
    if self._msg[2] is None:
```